packages/turbo-workspace-convert/__tests__/managers/npm.test.ts

```diff
@@ -0,0 +1,203 @@
+import fs from "fs-extra";
+import {
+  extendMatrix,
+  generateTestMatrix,
+  validateWorkspace,
+} from "../test-utils";
+
+const npm = MANAGERS.npm;
+
+describe("npm", () => {
+  const { useFixture } = setupTestFixtures({
+    directory: path.join(__dirname, "../../"),
+    test: "npm",
+  });
+
+  describe("detect", () => {
+    test.each([
+      ["npm", true],
+      ["yarn", false],
+      ["pnpm", false],
+    ])("detects npm workspaces from %s workspaces", async (from, result) => {
+      const { root } = useFixture({ fixture: `../${from}/basic` });
+
+      expect(
+        await npm.detect({
+          workspaceRoot: root,
+        })
+      ).toEqual(result);
+    });
+  });
+
+  describe("create", () => {
+    test.each(generateTestMatrix())(
+      "creates yarn workspaces from %s workspaces with interactive=%s and dry=%s",
+      async (from, interactive, dry) => {
+        const { root } = useFixture({ fixture: `../${from}/basic` });
+        const project = await MANAGERS[from].read({
+          workspaceRoot: root,
+        });
+
+        // convert to npm
+        await npm.create({
+          project,
+          to: { name: "npm", version: "8.19.2" },
+          logger: new Logger({ interactive, dry }),
+          options: {
+            interactive,
+            dry,
+          },
+        });
+
+        if (dry) {
+          expect(await MANAGERS[from].detect({ workspaceRoot: root })).toEqual(
+            true
+          );
+        } else {
+          expect(await npm.detect({ workspaceRoot: root })).toEqual(true);
+        }
+      }
+    );
+  });
+
+  describe("remove", () => {
+    test.each(
+      extendMatrix([
+        ["npm", "8.19.2", undefined],
+        ["yarn", "1.22.19", ["apps/*", "packages/*"]],
+        ["pnpm", "7.12.1", undefined],
+      ])
+    )(
+      "removes npm workspaces when moving to %s@%s | workspaces=%s withNodeModules=%s interactive=%s and dry=%s",
+      async (to, version, workspaces, withNodeModules, interactive, dry) => {
+        const { root, readJson } = useFixture({ fixture: "basic" });
+        if (withNodeModules) {
+          fs.ensureDirSync(path.join(root, "node_modules"));
+        }
+
+        const project = await npm.read({
+          workspaceRoot: root,
+        });
+
+        await npm.remove({
+          project,
+          to: { name: to, version },
+          logger: new Logger({ interactive, dry }),
+          options: {
+            interactive,
+            dry,
+          },
+        });
+
+        if (dry) {
+          expect(readJson(project.paths.packageJson).workspaces).toEqual(
+            project.workspaceData.globs
+          );
+        } else {
+          expect(readJson(project.paths.packageJson).workspaces).toEqual(
+            workspaces
+          );
+        }
+      }
+    );
+  });
+
+  describe("read", () => {
+    test.each<[PackageManagers, boolean]>([
+      ["yarn", true],
+      ["npm", false],
```

Comment:
I'd derive this, but don't care a lot.

Reply:
Yea I'm going to clean these up a bit more